### DISCUSSION TITLE
Removing all autobackup snapshots

### DISCUSSION THREAD

**Discussion by danboid** [psy0rz/zfs_autobackup#228] [Q&A] [answered]
I think I ended up staying up later than was wise trying to get autobackup set up and now  think I need to nuke all of my autobackup snapshots so that I can exclude a few datasets on my pool that weren't intended to be snapshot alongside the other datasets. I'm trying to destroy 2 datasets but I can't because they are rolled into the autobackup snapshots.

My understanding is that I should be able to run something like:

zfs set autobackup:mybackup=false mypool/dataset-I-dont-want-to-snapshot

Then I can delete all of the autobackup snapshots and then destroy these datasets I need to start over with.

I'm doing this because I am using --strip-path=1 so I presume setting theit autobackup setting to false is how it is dealt with if you don't want to replicate the entire pool? 

Is there a command or script I can run to make this process a bit easier? I will of course have to destroy all of the autoback snapshots on both the source and the destination machines but I've only got a few atm.

Thanks

**Comment by danboid** [comment]
I've used a command like this to generate a list of all the autoback snapshots on the source machine:

```
zfs list -r -H -t snapshot -o name | grep asteria-homedirs- > all-autoback-snaps
```
Where `asteria-homedirs` is the name of my autobackup ZFS label but its not letting me destroy any of the snapshots as root, I just get lots of errors like:

```
Failed to destroy astarray/home/age960@asteria-homedirs-20231012231530
cannot destroy snapshot astarray/home/age965@asteria-homedirs-20231012231530: dataset is busy
```

Is it possible for me to remove all of the autobackup snapshots without taking the server down, importing the pool into something like systemrescue-zfs or using the proxmox boot disc maybe and then deleting the snapshots like that? Why is the dataset "busy" when autobackup isn't even running?

Maybe I should be using zfs-autobackup to remove its own snapshots instead? How?

**Reply by psy0rz** [reply, marked as answer]
https://github.com/psy0rz/zfs_autobackup/wiki/Problems#dataset-is-busy

**Reply by danboid** [reply]
Yes, it was holds / tags stopping me destroying the autobackup snapshots.

After releasing the snapshots using the correct tag I was able to destroy them all.

**Comment by danboid** [comment]
For anyone else unfortunte enough to end up in this situation (please don't!), note that I had to remove all (well, most) of the datasets on the target machine in full after I had removed all of the autobackup snaphots and their holds on both ends in order to get zfs-autobackup working again. Ouch!

It wasn't the most fun thing I've ever had to go through but it was best to do it sooner rather than later.

**Comment by psy0rz** [comment]
As for commands to make things easier: pipe snapshot names to xargs -n1 zfs
